src/main/java/com/kodilla/ecommercee/controller/ProductController.java

```diff
@@ -1,39 +1,68 @@
 package com.kodilla.ecommercee.controller;
 
-
+import com.kodilla.ecommercee.domain.Product;
 import com.kodilla.ecommercee.domain.ProductDto;
+import com.kodilla.ecommercee.domain.ProductNotFoundException;
+import com.kodilla.ecommercee.domain.ProductsGroupNotFoundException;
+import com.kodilla.ecommercee.mapper.ProductMapper;
+import com.kodilla.ecommercee.service.ProductDbService;
+import lombok.RequiredArgsConstructor;
 import org.springframework.http.MediaType;
 import org.springframework.web.bind.annotation.*;
 
-import java.math.BigDecimal;
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
+@RequiredArgsConstructor
 @RestController
 @RequestMapping("/v1/product")
-public class
-ProductController {
-    @RequestMapping(method = RequestMethod.GET, value = "getProducts")
+public class ProductController {
+
+    private final ProductMapper productMapper;
+    private final ProductDbService productDbService;
+
+
+    @GetMapping(value = "getProducts")
     public List<ProductDto> getProducts() {
-        return new ArrayList<>();
+        return productMapper.mapToProductDtoList(productDbService.getAllProducts());
     }
 
     @GetMapping(value = "getProduct")
-    public ProductDto getProduct(@RequestParam Long productId) {
-        return new ProductDto(1L, "name", "description", new BigDecimal("100"), "1");
+    public ProductDto getProduct(@RequestParam Long productId) throws ProductNotFoundException {
+        return productMapper.mapToProductDto(
+                productDbService.getProductById(productId).orElseThrow(ProductNotFoundException::new));
     }
 
     @PostMapping(value = "createProduct", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public void createProduct(@RequestBody ProductDto productDto) {
+    public ProductDto createProduct(@RequestBody ProductDto productDto) throws ProductsGroupNotFoundException {
```

Comment:
Czy tu nie powinien być ProductNotFoundException?

Reply:
No nie, bo wyjątek rzuca metoda w Mapperze przy dobieraniu grupy na podstawie id z JSONA, i jest to ProductsGroupNotFoundException